### PR TITLE
Added optional predicate

### LINKED ISSUE #3: Boolean predicate for query
I think it would be very helpful if there would be an option to pass a function that returns a boolean value to the knn algorithm. If that is the case, the algorithm just considers items that satisfy that predicate.

What do people think about this idea?

### PR DESCRIPTION
**Pull request by AuspeXeu**
Added optional predicate which items need to satisfy in order to be cosidered for the result set

Resolves #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,7 +24,9 @@ function knn(tree, queryPoint, n) {
         }
 
         while (queue.length && queue.peek().isItem) {
-            result.push(queue.pop().node);
+            var node = queue.pop().node;
+            if (predicate(node))
+                result.push(node);
```
Comment (mourner) [inline index.js]: I'd prefer not calling an empty function if there's no predicate:

``` js
if (!predicate || predicate(node))
```

**Comment by mourner** [comment]
Also, needs a test.

File: index.js (hunk 1 above)
Comment (AuspeXeu) [inline index.js]: But this is already taken care of in the variable declaration, no? See line 11.

Comment (mourner) [inline index.js]: And I'm suggesting to remove it :)

Comment (AuspeXeu) [inline index.js]: This better? ;)

**Comment by AuspeXeu** [comment]
Added a test case.

**Comment by mourner** [comment]
:+1: thanks!

**Comment by mourner** [comment]
Released v1.1.0 with the change.

**Comment by AuspeXeu** [comment]
Thanks!

**Pull request merged**